### PR TITLE
Added Mutual information metric and least squares fitting.

### PR DESCRIPTION
**Pull request by AnejSvete**
- Added the Mutual Information metric and optimisation to the random module
- Added linear regression optimisation 
- Tests for the above

### CODE REVIEW

**Comment by AnejSvete** [comment]
The tests for mutual information are failing because of the large difference in the estimated mutual information for the true and the found vector, although the vectors are quite close. Not sure how to handle that yet.

File: latte/direction/common.py [hunk 1]
```diff
@@ -45,14 +46,17 @@ class MutualInformation(SimilarityMetric):
     """
 
     def score(self, projections: np.ndarray, scores: np.ndarray) -> float:
```
[the remark this answers is not in the document]
Comment (pawel-czyz) [reply latte/direction/common.py:48]: The implementation of this is in #4 (which also fixes the unit tests).

File: latte/direction/common.py [hunk 2]
```diff
@@ -94,7 +98,11 @@ def __init__(self) -> None:
 
 
 class MutualInformationMetric(ProjectionAxisMetric):
-    pass
+    """Mutual information metric between projections onto specified axis
+    and true generative factor scores."""
+
+    def __init__(self) -> None:
```
[the remark this answers is not in the document]
Comment (pawel-czyz) [reply latte/direction/common.py:104]: Same, it's in #4 .

**Review by pawel-czyz — APPROVED** [review]
I very much like your coding style! It's very clean and concise.

I have only noticed very minor things. Also, there's #4 which fixes the unit tests. I think it should be merged into this PR, and then this one can go into `main` :slightly_smiling_face:

File: latte/direction/common.py [hunk 3]
```diff
@@ -16,7 +17,7 @@ def score(self, projections: np.ndarray, scores: np.ndarray) -> float:
             scores: true generative factors values, shape (n_points,)
 
         Returns:
-            some measure of similarity between these two vectors, the larger the better
+            some measure of similarity between these two vectors, the, larger the better
```
[the remark this answers is not in the document]
Comment (AnejSvete) [reply latte/direction/common.py:20]: Sorry! Pycharm was suggesting this to me constantly! :upside_down_face:

**Pull request merged**